client/src/Views/Home/Campaigns/Campaign/CampaignProspectsStepBar.js

```diff
@@ -0,0 +1,145 @@
+import React, { useEffect, useState } from "react";
+import {
+  makeStyles,
+  Table,
+  TableHead,
+  TableRow,
+  TableCell
+} from "@material-ui/core";
+import colors from "Components/Styles/Colors";
+
+const useStyles = makeStyles({
+  root: {
+    width: "100%"
+  },
+  stepButton: {
+    color: `${colors.black}`,
+    fontSize: ".8rem",
+    fontWeight: "bold",
+    padding: "10px 0px 10px 0px",
+    height: ".8rem",
+    minWidth: "75px",
+    maxWidth: "150px",
+    textAlign: "center",
+    borderRight: `1px solid ${colors.borderGray}`,
+    borderBottom: `1px solid ${colors.borderGray}`,
+    borderTop: `1px solid ${colors.borderGray}`,
+    borderSpacing: "0",
+    borderCollapse: "separate",
+    "&:hover": {
+      backgroundColor: `${colors.lightGreen}`,
+      color: `${colors.white}`
+    },
+    "&:first-child": {
+      borderLeft: `1px solid ${colors.borderGray}`,
+      borderTopLeftRadius: "6px",
+      borderBottomLeftRadius: "6px"
+    },
+    "&:last-child": {
+      borderRadius: "0 7px 7px 0"
+    }
+  },
+
+  stepButtonColored: {
+    backgroundColor: `${colors.lightGreen}`,
+    color: "#ffffff",
+    fontWeight: "bold",
+    maxWidth: "150px",
+    fontSize: ".8rem",
+    height: ".8rem",
+    padding: "10px 0px 10px 0px",
+    minWidth: "75px",
+    textAlign: "center",
+    borderRight: `1px solid ${colors.borderGray}`,
+    borderBottom: `1px solid ${colors.borderGray}`,
+    borderTop: `1px solid ${colors.borderGray}`,
+    borderSpacing: "0",
+    borderCollapse: "separate",
+    "&:first-child": {
+      borderLeft: `1px solid ${colors.borderGray}`,
+      borderTopLeftRadius: "6px",
+      borderBottomLeftRadius: "6px"
+    },
+    "&:last-child": {
+      borderRadius: "0 7px 7px 0"
+    }
+  },
+
+  stepBar: {
+    borderCollapse: "separate",
+    fontSize: ".8rem",
+    height: ".8rem",
+    padding: "10px 0px 10px 0px"
+  }
+});
+
+const defaultStepOptions = ["All", "Pending"];
+
+const CampaignProspectsStepBar = props => {
+  const classes = useStyles();
+  const [stepBarArray, setStepBarArray] = useState([]);
+  const [recentlyFetched, setRecentlyFetched] = useState(false);
+  const [selectedButton, setSelectedButton] = useState([]);
+
+  useEffect(() => {
+    if (props.steps.length > 0 && !recentlyFetched) {
+      let result = props.steps.map(step => step.name);
+      result = defaultStepOptions.concat(result);
+      setStepBarArray(result);
+      let buttonActiveArray = [];
+      if (selectedButton.length > 0) {
+        buttonActiveArray = selectedButton;
+      } else {
+        for (let i = 0; i < result.length; i++) {
+          if (i === 0) {
+            buttonActiveArray.push(true);
+          } else {
+            buttonActiveArray.push(false);
+          }
+        }
+      }
+      setSelectedButton(buttonActiveArray);
+      setRecentlyFetched(true);
+    }
+  });
```

Comment:
Again, you are using a similar flow to the component above, but this is not that efficient, there is probably a way to write this without this useEffect. For example, instead of having a state variable called stepBarArray, you can just have that data generated in your render function based on props.step. In essence, you can use props.step to generate the step bar, instead of overusing state by setting it based on the props